webroot/adm/keysets.html

```diff
@@ -78,22 +110,44 @@ <h3>Output</h3>
         const jo = JSON.parse(text);
         let allowlist = jo.allowlist;
 
-        allowlist = allowlist.filter((value, _, __) => !remove.includes(value))
-        allowlist = allowlist.concat(add.filter((value, _, __) => !allowlist.includes(value)))
-
         const url = '/api/sharing/keyset';
 
+        let payload = {keyset_id: keysetId}
+
+        if((add && add.length !== 0) || (remove && remove.length !== 0)) {
+          allowlist = allowlist.filter((value, _, __) => !remove.includes(value))
+          allowlist = allowlist.concat(add.filter((value, _, __) => !allowlist.includes(value)))
+        }
+
+        payload["allowlist"] = allowlist
 
-        doApiCall('POST', url, '#standardOutput', 'errorOutput', JSON.stringify({allowlist: allowlist, keyset_id: keysetId, site_id: siteId}))
+        if(name && (name = $.trim(name))) {
+          payload["name"] = name
+        }
+
+        doApiCall('POST', url, '#standardOutput', '#errorOutput', JSON.stringify(payload))
       }, (err) => {
-        $('#errorOutput').html('Error: ' + err.status + ': ' + err.statusText);
+        $('#errorOutput').html('Error: ' + err.status + ': ' + (isJsonString(err.responseText) ? JSON.parse(err.responseText).message : (err.responseText ? err.responseText : err.statusText)));
       });
 
     });
     $('#doMakeNew').on('click', function () {
       const siteId = parseInt($('#siteId').val());
+      if(!siteId) {
+        $('#errorOutput').html("Required Parameters: site_id")
+        return;
+      }
+      let name = $('#keysetName').val();
       const url = '/api/sharing/keyset';
-      doApiCall('POST', url, '#standardOutput', 'errorOutput', JSON.stringify({allowlist: [] ,site_id: siteId}));
+
+      let payload;
+      if(name && (name = $.trim(name))) {
+        payload = JSON.stringify({allowlist: [], site_id: siteId, name: name})
```

Comment:
@jon8787 @zaiweidu when we create new keyset, should we allow people to provide a allowlist straight away instead of making adding allowlist as a separate second request?

Reply:
Re "allowed_sites straight away", let's solve this later when we offer multiple keysets. For now, people won't generally be adding keysets since the first one is created automatically.